### PR TITLE
Make postgres_backend use generic IO type

### PR DESCRIPTION
**Pull request by petuhovskiy**
## Describe your changes

- Support measuring inbound and outbound traffic in MeasuredStream
- Start using MeasuredStream in safekeepers code

## Issue ticket number and link

Preliminary traffic measuring support for https://github.com/neondatabase/neon/issues/3200

## Checklist before requesting a review
- [X] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## PR details

`PostgresBackendTCP` is created for convenience, to use in code that doesn't need support for non-TCP connections and doesn't want to carry generic parameters around.

Advice on how to make generic parameters more pretty is welcome.

In the next PR, I'm going to add more labels to `PG_IO_BYTES` metric, to actually measure cross-az traffic.

### CODE REVIEW

File: libs/postgres_backend/src/lib.rs [hunk 1]
```diff
@@ -288,6 +290,25 @@ impl PostgresBackend {
             peer_addr,
         })
     }
+}
+
+impl<IO: AsyncRead + AsyncWrite + Unpin> PostgresBackend<IO> {
+    pub fn new_from_io(
```
Comment (arssher) [inline libs/postgres_backend/src/lib.rs:296]: it can be merged with `fn new()`

**Review by arssher — APPROVED** [review]

**Comment by arssher** [comment]
Let's mentioned added metrics in commit message.

File: libs/postgres_backend/src/lib.rs (hunk 1 above)
Comment (petuhovskiy) [reply libs/postgres_backend/src/lib.rs:296]: Do you mean call `new_from_io()` inside `new()`, or remove `new()` altogether (then we will need to always pass peer_addr)?

**Comment by petuhovskiy** [comment]
> Let's mentioned added metrics in commit message.

I will mention them and add a test later. Right now they are PoC and will be reworked in the PR (new labels will be added).

File: libs/postgres_backend/src/lib.rs (hunk 1 above)
Comment (arssher) [reply libs/postgres_backend/src/lib.rs:296]: Ah, I forgot about peer_addr. Then either variant is ok, including current.

**Pull request merged**